platforms/macos/src/subclass.rs

```diff
@@ -33,7 +33,7 @@ static SUBCLASSES: SyncLazy<Mutex<HashMap<&'static AnyClass, &'static AnyClass>>
 static mut ASSOCIATED_OBJECT_KEY: u8 = 0;
 
 fn associated_object_key() -> *const c_void {
-    unsafe { &ASSOCIATED_OBJECT_KEY as *const u8 as *const _ }
+    unsafe { std::ptr::addr_of_mut!(ASSOCIATED_OBJECT_KEY) as *const u8 as *const _ }
```

Comment:
The `unsafe` block can be removed entirely, and the cast replaced with stricter `pointer::cast`:

```suggestion
    std::ptr::addr_of!(ASSOCIATED_OBJECT_KEY).cast::<u8>();
```

Reply:
I tried your suggestion and got this compiler error:

error[E0133]: use of mutable static is unsafe and requires unsafe block

Reply:
Sorry for the incorrect information — it seems I incorrectly assumed that since no _reference_ was being taken, `addr_of!` would be safe.